pkg/storage/streams/store_test.go

```diff
@@ -0,0 +1,120 @@
+package streams
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"storj.io/storj/pkg/paths"
+	ranger "storj.io/storj/pkg/ranger"
+	"storj.io/storj/pkg/storage/segments"
+)
+
+var ctx = context.Background()
+
+type metaClosure = func(context.Context, paths.Path) (segments.Meta, error)
+type getClosure = func(context.Context, paths.Path) (ranger.Ranger, segments.Meta, error)
+type putClosure = func(context.Context, paths.Path, io.Reader, []byte, time.Time) (segments.Meta, error)
+type deleteClosure = func(context.Context, paths.Path) error
+type listClosure = func(context.Context, paths.Path, paths.Path, paths.Path, bool, int, uint32) ([]segments.ListItem, bool, error)
+
+type segmentStub struct {
+	mc metaClosure
+	gc getClosure
+	pc putClosure
+	dc deleteClosure
+	lc listClosure
+}
+
+func (s *segmentStub) Meta(ctx context.Context, path paths.Path) (meta segments.Meta, err error) {
+	return s.mc(ctx, path)
+}
+
+func (s *segmentStub) Get(ctx context.Context, path paths.Path) (rr ranger.Ranger,
+	meta segments.Meta, err error) {
+	return s.gc(ctx, path)
+}
+
+func (s *segmentStub) Put(ctx context.Context, path paths.Path, data io.Reader, metadata []byte,
+	expiration time.Time) (meta segments.Meta, err error) {
+	return s.pc(ctx, path, data, metadata, expiration)
+}
+func (s *segmentStub) Delete(ctx context.Context, path paths.Path) (err error) {
+	return s.dc(ctx, path)
+}
+func (s *segmentStub) List(ctx context.Context, prefix, startAfter, endBefore paths.Path,
+	recursive bool, limit int, metaFlags uint32) (items []segments.ListItem,
+	more bool, err error) {
+	return s.lc(ctx, prefix, startAfter, endBefore, recursive, limit, metaFlags)
+}
+
+var segmentErrorString = "a segment error"
+var protoErrorString = "proto: streams.MetaStreamInfo: wiretype end group for non-group"
+
+func TestMeta(t *testing.T) {
+
+	errorFn := func(ctx context.Context, path paths.Path) (segments.Meta, error) {
+		return segments.Meta{}, errors.New(segmentErrorString)
+	}
+	emptyFn := func(ctx context.Context, path paths.Path) (segments.Meta, error) {
+		return segments.Meta{
+			Modified:   time.Time{},
+			Expiration: time.Time{},
+			Size:       0,
+			Data:       []byte(""),
+		}, nil
+	}
+	metaFn := func(ctx context.Context, path paths.Path) (segments.Meta, error) {
+		return segments.Meta{
+			Modified:   time.Now(),
+			Expiration: time.Now(),
+			Size:       10,
+			Data:       []byte("data"),
+		}, nil
+	}
+
+	metaSlice := []metaClosure{
+		errorFn,
+		emptyFn,
+		metaFn,
+	}
+
+	pathSlice := []paths.Path{
+		paths.New(""),
+		paths.New("bucket"),
+	}
+
+	for i, closure := range metaSlice {
```

Comment:
for the table driven development, it would be good to have a range over some sort of test-specific struct. one field would be the closure, but other fields would be what the test thinks is expected behavior. it looks like right now you're not actually checking anything about the return value in the non-error case. (there are no checks if err == nil). 

if you had a small struct here with some test-specific fields, you could say that the errorFn is expected to return no values, the emptyFn test should make sure that it returns successfully but no metadata, and the metaFn test returns metadata successfully.

more to the point - you could actually define the closures themselves in terms of these test expectations inside of the for loop.

so, 

```golang
type testStruct struct {
  segmentErr error
  size int64
  segmentMetaData []byte
}
for _, test := range []testStruct{
  {errors.New(segmentErrorString), 0, nil},
  {nil, 0, ""},
  {nil, 10, "data"},
 } {
   ...
   // define the streamstore and closure in here, do the test based on what the testStruct says to do
}
```